src/com/jalasoft/search/model/FactoryFileObject.java

```diff
@@ -25,10 +30,38 @@
  */
 public class FactoryFileObject {
 
-
-    public static void setFileObject(FileObject file)
+    public static void setFileObjectFound(SearchCriterial searchCriterial, String typeOfFile, String fileName,
+                                          String fileDirectory, boolean readOnly, boolean hidden, String ownerName,
+                                          Date dateModified, long size, Date dateCreated, Date dateAccessed,
+                                          String fileContains, int numberOfFiles)
     {
-
-
+        if (typeOfFile.equalsIgnoreCase("file"))
+        {
+            searchCriterial.setFileFound(new com.jalasoft.search.model.File(FilenameUtils.getExtension(fileName), fileContains));
```

Comment:
@mpardo84 The typeObject is used to compare with the extension obtained by FilenameUtils.getExtension(fileName) that is the File extension needed by the FileObject so it is not needed in the Factory. If you want, I can change the name of "typeOfFile" to another.